File: Odd_Even_or_Cheating.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int p, j1, j2, r, a;
    cin >> p >> j1 >> j2 >> r >> a;

    if (r) cout << (a ? "Jogador 2 ganha!\n" : "Jogador 1 ganha!\n");
    else if (a) cout << "Jogador 1 ganha!\n";
    else {
        int sum = (j1 + j2) % 2;
        if ((sum == 0 && p == 1) || (sum == 1 && p == 0))
            cout << "Jogador 1 ganha!\n";
        else
            cout << "Jogador 2 ganha!\n";
    }
}

/* #include <bits/stdc++.h>
using namespace std;

int main() {
    int p, j1, j2, r, a;
    cin >> p >> j1 >> j2 >> r >> a;

    // cheat case
    if (r == 1) {
        if (a == 1) cout << "Jogador 2 ganha!\n";
        else cout << "Jogador 1 ganha!\n";
    }
    else { // no cheat
        if (a == 1) {
            cout << "Jogador 1 ganha!\n";
        }
        else {
            int sum = j1 + j2;

            if (sum % 2 == 0) { // even
                if (p == 1) cout << "Jogador 1 ganha!\n";
                else cout << "Jogador 2 ganha!\n";
            } else { // odd
                if (p == 0) cout << "Jogador 1 ganha!\n";
                else cout << "Jogador 2 ganha!\n";
            }
        }
    }

    return 0;
} */

File: Odd_Even_or_Cheating.py
p, j1, j2, r, a = map(int, input().split())

if r == 1:
    if a == 1:
        print("Jogador 2 ganha!")
    else:
        print("Jogador 1 ganha!")
else:
    if a == 1:
        print("Jogador 1 ganha!")
    else:
        s = j1 + j2
        if s % 2 == 0:  # even
            if p == 1:
                print("Jogador 1 ganha!")
            else:
                print("Jogador 2 ganha!")
        else:  # odd
            if p == 0:
                print("Jogador 1 ganha!")
            else:
                print("Jogador 2 ganha!")